### ISSUE TITLE
got a System.OutOfMemoryException

### ISSUE THREAD

**Issue by pinopino** [helios-io/helios#59]
i'm interest in helios. but i got a `System.OutOfMemoryException` when i ran TimeService example on single local machine. sorry i can't support the stacktrace info, but below is the method throw this exception (it's in helios-1.4.1\src\Helios\Concurrency\Impl\DedicatedThreadPoolFiber.cs):
```
public void Add(Action op)
{
     if (Running)
         _threadPool.QueueUserWorkItem(() => Executor.Execute(op));
}
```
os: window 7 ultimate 64 bit
memory: 16g

**Comment by kklldog** [comment]
me too.....

**Comment by samcov** [comment]
I too have seen this! I first saw it in Akka's TimeService sample, and wondered if it was Actor related.

I later tried Helios's version and the same problem occurred.

This is obviously a Helios issue, and it's a real show stopper!

**Comment by samcov** [comment]
Just noticed that Helios is depreciated in favor of DotNetty from Azure.

**Comment by Aaronontheweb** [comment]
#75 related - looking into it.

**Comment by Aaronontheweb** [comment]
99% sure the issue here is the unnecessary allocation of new delegates.... This might be a 1-line fix. Let's see.

**Comment by Aaronontheweb** [comment]
Fixed as part of 2.1 using the new APIs.
